movies-manager/src/components/movie-card/movie-card.component.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faAlignJustify, faTimes } from "@fortawesome/free-solid-svg-icons";
+import { IMovie } from "../../types/types";
+import { menuItemTitles } from "./mockMenuTitles";
+import "./movie-card.component.scss";
+
+const blockName = 'movie';
+
+interface IMovieCardState {
+	isEditMenuVisible: boolean;
+}
+
+interface IMovieCardProps {
+	movie: IMovie;
+    onClickMovie: (modalDialogType: any, id: number, isOpen: boolean) => void
+}
+
+export class MovieCard extends Component<IMovieCardProps, IMovieCardState> {
+    private wrapperRef: React.RefObject<HTMLInputElement>;
+
+    constructor(props: IMovieCardProps) {
+        super(props);
+        this.wrapperRef = React.createRef();
+        this.state = {
+            isEditMenuVisible: false,
+        };
+    }
+
+    handleClickOutside = (event: Event) => {
+        if (this.wrapperRef && !this.wrapperRef.current.contains(event.target as Node)) {
+            this.hideEditMenu();
+        }
+    }
+
+    showEditMenu = () => {
+        this.setState({ isEditMenuVisible: true });
+        document.addEventListener('mousedown', this.handleClickOutside);
```

Comment:
You are using React to work with a DOM. Find a way how to do the same without document.addEventListener.

Reply:
done